### PR TITLE
Add section to the custom field documentation on ORM filtering.

### PR DESCRIPTION
**Pull request by Kircheneer**
# Closes: DNE
# What's Changed

Added a small section to the custom field documentation that explains, how you can use the ORM to filter on custom field values.

### CODE REVIEW

**Comment by Kircheneer** [comment]
I have used the PR number for the changelog fragment as there is no corresponding issue - please let me know if you do need an issue and I will create one.

File: nautobot/docs/models/extras/customfield.md [hunk 1]
```diff
@@ -68,6 +68,15 @@ There are a number of available built-in filters for custom fields.
 
 Filtering on an object's list view follows the same pattern as [custom field filtering on the API](../../rest-api/filtering.md#filtering-by-custom-field).
 
+When using the ORM, you can filter on custom fields as follows (assuming a custom field of string type called 'site_code' was created for Site objects):
+
+```python
+from nautobot.dcim.models import Site
+
+all_sites = Site.objects.all()  # -> ['Raleigh', 'Charlotte']
+filtered_sites = Site.objects.filter(_custom_field_data__mycustomfield="US-NC-RAL42")  # -> ['Raleigh']
```
Comment (itdependsnetworks) [inline nautobot/docs/models/extras/customfield.md:77]: Shouldn’t it be site_code? Based on the prior description

Comment (Kircheneer) [reply nautobot/docs/models/extras/customfield.md]: That is of course correct - fixed.

File: changes/2431.changed [hunk 2]
```diff
@@ -0,0 +1 @@
+Add section to the custom field documentation on ORM filtering.
```
Comment (glennmatthews) [inline changes/2431.changed:1]: Nit: Possibly `2431.added` since this is purely new content.

File: nautobot/docs/models/extras/customfield.md [hunk 3]
```diff
@@ -68,6 +68,15 @@ There are a number of available built-in filters for custom fields.
 
 Filtering on an object's list view follows the same pattern as [custom field filtering on the API](../../rest-api/filtering.md#filtering-by-custom-field).
 
+When using the ORM, you can filter on custom fields as follows (assuming a custom field of string type called 'site_code' was created for Site objects):
```
Comment (glennmatthews) [inline nautobot/docs/models/extras/customfield.md:71]:
```suggestion
When using the ORM, you can filter on custom fields using `_custom_field_data__<field name>` (note the underscore before `custom_field_data` and the double-underscore before the field name). For example, if a custom field of string type with a `name` of  `"site_code"` was created for Site objects, you could filter as follows:
```

File: nautobot/docs/models/extras/customfield.md [hunk 4]
```diff
@@ -68,6 +68,15 @@ There are a number of available built-in filters for custom fields.
...
+
+```python
+from nautobot.dcim.models import Site
+
+all_sites = Site.objects.all()  # -> ['Raleigh', 'Charlotte']
+filtered_sites = Site.objects.filter(_custom_field_data__site_code="US-NC-RAL42")  # -> ['Raleigh']
```
Comment (glennmatthews) [inline nautobot/docs/models/extras/customfield.md:77]: Perhaps build on the example a bit:

```suggestion
all_sites = Site.objects.all()  # -> ['Raleigh', 'Charlotte', 'Greensboro']
filtered_sites_1 = Site.objects.filter(_custom_field_data__site_code="US-NC-RAL42")  # -> ['Raleigh']
filtered_sites_2 = Site.objects.filter(_custom_field_data__site_code__in=["US-NC-RAL42", "US-NC-CLT22"])  # -> ['Raleigh', 'Charlotte']
```

**Review by glennmatthews — APPROVED** [review]
Good stuff!

**Pull request merged**